src/CTe/DacteV3.php

```diff
@@ -2798,6 +2798,7 @@ protected function zModalAquaviario($x = 0, $y = 0)
                 ->item(0)->nodeValue;
         } elseif ($this->infNFe->item(0) !== null
             && $this->infNFe->item(0)->getElementsByTagName('infUnidCarga') !== null
+            && $this->infNFe->item(0)->getElementsByTagName('infUnidCarga')->item(0) !== null
```

Comment:
Acredito que a mesma condição também poderia ser aplicada para o `infNF` e `infOutro`?

Reply:
Na verdade, o certo é a documentação na nota técnica. Se esse campo é obrigatório nos outros então já está certo e não irá acontecer `Call to a member function getElementsByTagName() on null.` Confirme na NT. :smile:

Reply:
Precisa de alguma alteração no PR?

Reply:
Sim, pois de acordo com o nfephp-org/sped-cte/schemas/PL_CTe_300/cteTiposBasico_v3.00.xsd infUnidCarga existe dentro de infOutros, infNFe e infNF e como a mínima ocorrência é 0 defensivamente é bom adicionar nas outras condicionais.

Entretanto, não trabalho com o modal aquaviário então confirme (modal aéreo ainda está TODO) se realmente a informação a ser exibida dentro de IDENTIFICAÇÃO DOS CONTEINERS deve ser Informações das Unidades de Carga. Levanto essa questão pois fiquei em dúvida quando li a NotaTécnica 2016_002.pdf (
[NotaTécnica 2016_002.pdf](https://github.com/nfephp-org/sped-da/files/1808032/NotaTecnica.2016_002.pdf)
) página 18 INFORMAÇÕES ESPECÍFICAS DO MODAL AQUAVIÁRIO. Talvez algum contribuidor também possa ajudar.

Reply:
Vou alterar aqui e mando